src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/PreloginSubHandlers/TlsAuthenticator.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Security;
+using System.Security.Authentication;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.Data.SqlClient.SNI;
+using Microsoft.Data.SqlClientX.IO;
+
+namespace Microsoft.Data.SqlClientX.Handlers.Connection.PreloginSubHandlers
+{
+    internal class TlsAuthenticator
+    {
+        public virtual async ValueTask AuthenticateClientInternal(PreloginHandlerContext request, SslClientAuthenticationOptions options, bool isAsync, CancellationToken ct)
+        {
+            try
+            {
+                await AuthenticateClient(request, options, isAsync, ct).ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                // TODO: Convert to a Sql Exception.
+                // this would require that we convert the error to a Sql error with the traditional details about 
+                // SNI. A lot of errors strings require SNI providers to be passed in.
+                // So we will stick to the current format in Managed SNI, for creating these exceptions.
+                throw;
+            }
+
+            LogWarningIfNeeded(request);
+
+            static void LogWarningIfNeeded(PreloginHandlerContext request)
+            {
+                string warningMessage = request.ConnectionContext.SslStream.SslProtocol.GetProtocolWarning();
+                if (!string.IsNullOrEmpty(warningMessage))
+                {
+                    if (ShouldNotLogWarning(request))
+                    {
+                        // Skip console warning
+                        SqlClientEventSource.Log.TryTraceEvent("<sc|{0}|{1}|{2}>{3}",
+                            nameof(TlsAuthenticator),
+                            nameof(AuthenticateClientInternal),
+                            SqlClientLogger.LogLevel.Warning,
+                            warningMessage);
+                    }
+                    else
+                    {
+                        // This logs console warning of insecure protocol in use.
+                        request.ConnectionContext.Logger.LogWarning(nameof(TlsAuthenticator), nameof(AuthenticateClientInternal), warningMessage);
+                    }
+                }
+            }
+
+            static bool ShouldNotLogWarning(PreloginHandlerContext request)
```

Comment:
In the initial code it was an inline condition.

The reason I put this as a local function was so that the code reads well. Right now I can see that I am evaluating the condition to check whether it should log warning or not. 
However when you inline it or make a variable out of it, the readability take a hit.